dotnet/src/Skills/Skills.OpenAPI/OpenApi/OpenApiDocumentParser.cs

```diff
@@ -138,9 +138,9 @@ private async Task<JsonObject> DowngradeDocumentVersionToSupportedOneAsync(Strea
 
         var obj = serializer.Deserialize(stream);
 
-        using var memoryStream = new MemoryStream(Encoding.UTF8.GetBytes(JsonSerializer.Serialize(obj)));
+        using var memoryStream = new MemoryStream(Encoding.UTF8.GetBytes(JsonSerializer.Serialize(obj, obj!.GetType(), SourceGenerationContext.Default)));
```

Comment:
> Are there more types that might be used here?

Likely.  There are no additional warnings produced, and I only discovered these specific types by iterating on the tests until things passed.

Reply:
> and I only discovered these specific types by iterating on the tests until things passed.

That's why I asked how robust the tests are. This is potentially a breaking change even if someone doesn't use trimming. The Reflection based serializer would have "just worked" before, but now only the above types will be deserialized. If the tests are exhaustive, then I think we are fine.

Reply:
> If the tests are exhaustive, then I think we are fine.

They're not "exhaustive".  I don't know if they're "exhaustive enough".

So what would you recommend?  Just from looking at the SharpYaml code, which is no longer actively maintained btw, it can produce many things given the right input.  I can speculatively add a bunch of primitives, but that's not going to help with, for example, enum values.

Seems the answers are either a) stop using this particular dependency (which isn't something I can do here), b) ensure there's a reflection-based fallback in place for when not using native aot, c) just say this whole library isn't native aot friendly, or d) hope for the best.

Reply:
> So what would you recommend? 

I'm not 100% sure yet, still trying to gather information. Doing a quick check, if we marked this method as `[RequiresUnreferenced/DynamicCode]`, I _think_ it would only affect these 2 public APIs:

https://github.com/microsoft/semantic-kernel/blob/45245f91f55d8cc8d8a9bd514c82f3be17da38e1/dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelChatGptPluginExtensions.cs#L113-L174

We could just mark them as `[RequiresUnreferenced/DynamicCode]` for now, telling users that this may not work with trimming/AOT. And then later, when we have time to ensure the YAML and JSON changes work with trimming/AOT, we can remove the attributes.

Reply:
> I think it would only affect these 2 public APIs

I pushed up a commit with the change.  I had to attribute 13 methods with it to satisfy all the warnings; 10 are public.

Is this the right answer?

Reply:
> Is this the right answer?

IMO - I think so. As I put in the other review, I think we could just revert the source generator usage here now that we have the attribute on the methods. That way we don't change behavior here.

Reply:
> That way we don't change behavior here.

Isn't there a way to say "fall back to using reflection if it's available and you like the necessary type info otherwise"?

Reply:
> fall back to using reflection if it's available and you lack the necessary type info otherwise

One idea is to chain the `DefaultJsonTypeInfoResolver` into the `TypeInfoResolver` of the `JsonSerializerOptions` passed in. You wouldn't be using the `JsonTypeInfo`. Instead you'd be using the JsonSerializerOptions overload.

Reply:
That would require either mutating the existing options, which isn't allowed, or creating a new one on each call, which we say is really bad. No? This was also discussed in that teams discussion.

Reply:
Sorry, I thought you were commenting on a different options discussion, the one inside of the converter. Misread on my phone. I'll reread your suggestion later.

Reply:
DefaultJsonTypeInfoResolver was introduced in .NET 7.  This is still using the 6.0 S.T.J.  And there are resolution conflicts that currently arise when trying just to update S.T.J to 7.0.

Reply:
I think what you have now is the right call for now. This can be revisited when someone needs it to work with trimming/AOT.

Reply:
(I'm not sure how to "resolve" this conversation now that it is "Outdated". If you figure out how, please resolve it. 😄)